### PR TITLE
Fixes 280: switch to using repomd checksum

### PR DESCRIPTION
**Pull request by jlsherrill**
to determine if a repo was updated.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMSCONTENT-280

**Comment by jlsherrill** [comment]
To test this, you can use this repository url: https://jlsherrill.fedorapeople.org/fake-repos/revision/current/

You should see one package.  Ask me to 'switch' it and add a 2nd package.  Then introspect it again, or wait ~8 hours.  you should now see 2 packages

**Review by Andrewgdewar — APPROVED** [review]
Tested for functionality. Works as intended. 
Code looks good but will defer to @rverdile's opinion in that regard.
Ack

File: pkg/external_repos/introspect.go [hunk 1]
```diff
@@ -84,7 +86,13 @@ func Introspect(repo *dao.Repository, repoDao dao.RepositoryDao, rpm dao.RpmDao)
 		return 0, err
 	}
 
-	if repo.Revision != "" && repomd.Revision != "" && repomd.Revision == repo.Revision {
+	checksumStr := ""
+	if repomd.RepomdString != nil && *repomd.RepomdString != "" {
+		sum := sha256.Sum256([]byte(*repomd.RepomdString))
+		checksumStr = hex.EncodeToString(sum[:])
```
Comment (rverdile) [inline pkg/external_repos/introspect.go:92]: Does this package from the standard library accomplish the same thing? https://pkg.go.dev/encoding/hex#EncodeToString

Also, just curious, why do we need to encode the checksum here?

Comment (jlsherrill) [reply pkg/external_repos/introspect.go:92]: doh! thats what i meant to use.  IDE autocomplete for the lose. 

The checksum comes through as a byte array of numbers.  We could just store those.. but converting it to hex matches what you normally see when you compute a sha256 checksum

**Review by mshriver — APPROVED** [review]

**Pull request merged**